test/utils/consoleError.js

```diff
@@ -0,0 +1,17 @@
+/* eslint-disable no-console */
+// Makes sure the tests fails when a PropType validation fails.
+function consoleError() {
+  console.error = (...args) => {
+    // Can't use log as karma is not displaying them.
+    console.info(...args);
+    throw new Error(...args);
```

Comment:
These shouldn't be re-introduced. They were removed intentionally because they're highly problematic which is explained extensively in https://github.com/mui-org/material-ui/pull/21581.

Reply:
From my perspective, failing the CI when an console.error is thrown even if it's hard to identify the origin of is still a step forward compared to silently ignoring it. In our case study, we could easily find the origin. Do you have an alternative solution in mind?

Reply:
To be sure we are on the same page, #21581 ignores Karma, this pull request focus on Karma only. Are we agreement on this fact?

Reply:
Yep but since this repository has not the issues you describe I don't want to re-introduce a bad approach.

Reply:
How does this repository doesn't have the issue? Shouldn't this fail https://github.com/mui-org/material-ui/pull/22075/commits/bda2d77ccfdce48c45c807dea14285669103ff31?

Reply:
> How does this repository doesn't have the issue? Shouldn't this fail bda2d77? It's green scream!

Yes for your highly constructed case it is not ideal. Practically this never happened.

Reply:
> Also, I don't think that which repository it appears into matter since it uses the same logic.

It matters a lot. We wouldn't be able to move at all if we consider all repositories.

Reply:
The Ok case does not exist in praxis which is what I'm intersted in. Creating a repository that can be worked on not one that looks good on paper.